src/platform/packages/shared/shared-ux/chrome/navigation/src/ui/components/navigation_section_ui.tsx

```diff
@@ -402,7 +403,8 @@ function nodeToEuiCollapsibleNavProps(
       isSelected,
       onClick,
       icon: navNode.icon,
-      title: navNode.title,
+      // @ts-ignore title accepts JSX elements and they render correctly but the type definition expects a string
+      title: !subItems && navNode.withBadge ? <SubItemTitle item={navNode} /> : navNode.title,
```

Comment:
This should be fixed in https://github.com/elastic/eui/blob/7ea5ed4828e371529e39ecd95abe74aa52fa8e89/packages/eui/src/components/collapsible_nav_beta/collapsible_nav_item/collapsible_nav_item.tsx#L52. For now, I would suggest:
```suggestion
      // @ts-expect-error title accepts JSX elements and they render correctly but the type definition expects a string
      title: !subItems && navNode.withBadge ? <SubItemTitle item={navNode} /> : navNode.title,
```